src/Sarif.Viewer.VisualStudio/ErrorList/ErrorListService.cs

```diff
@@ -268,30 +299,16 @@ private static string PromptForFileSaveLocation(string dialogTitle, string input
                 null;
         }
 
-        private static void SaveLogFile(string filePath, SarifLog log)
+        private static async System.Threading.Tasks.Task SaveLogFileAsync(string filePath, string logText)
         {
-            // For now this is being done on the UI thread
-            // and is only required due to the message box being shown below.
-            // This will be addressed when https://github.com/microsoft/sarif-visualstudio-extension/issues/160
-            // is fixed.
-            ThreadHelper.ThrowIfNotOnUIThread();
-
-            SaveLogFile(filePath, JsonConvert.SerializeObject(log));
-        }
-
-        private static void SaveLogFile(string filePath, string logText)
-        {
-            // For now this is being done on the UI thread
-            // and is only required due to the message box being shown below.
-            // This will be addressed when https://github.com/microsoft/sarif-visualstudio-extension/issues/160
-            // is fixed.
-            ThreadHelper.ThrowIfNotOnUIThread();
-
             string error = null;
 
             try
             {
-                File.WriteAllText(filePath, logText);
+                using (StreamWriter streamWriter= File.CreateText(filePath))
+                {
+                    await streamWriter.WriteAsync(logText).ConfigureAwait(continueOnCapturedContext: false);
+                }
```

Comment:
Just `File.WriteAllTextAsync`?

Reply:
If that existed I would use it :)

---
In reply to: [494573646](https://github.com/microsoft/sarif-visualstudio-extension/pull/202#discussion_r494573646) [](ancestors = 494573646)